packages/eas-cli/src/metadata/context.ts

```diff
@@ -0,0 +1,110 @@
+import { App, Session } from '@expo/apple-utils';
+import { ExpoConfig } from '@expo/config';
+import { Platform } from '@expo/eas-build-job';
+import { IosSubmitProfile } from '@expo/eas-json/build/submit/types';
+import assert from 'assert';
+
+import { CredentialsContext } from '../credentials/context';
+import { getRequestContext } from '../credentials/ios/appstore/authenticate';
+import { getExpoConfig } from '../project/expoConfig';
+import { getBundleIdentifierAsync } from '../project/ios/bundleIdentifier';
+import { Actor } from '../user/User';
+import { ensureLoggedInAsync } from '../user/actions';
+import { getProfilesAsync } from '../utils/profiles';
+
+export type MetadataContext = {
+  /** Submission profile platform to use */
+  platform: Platform.IOS;
+  /** Submission profile name to use */
+  profile: IosSubmitProfile;
+  /** Configured store configuration file name (defaults to store.config.json) */
+  metadataFile: string;
+  /** Authenticated Expo account */
+  user: Actor;
+  /** The store credentials manager */
+  credentialsCtx: CredentialsContext;
+  /** The app bundle identifier to use for the stores */
+  bundleIdentifier: string;
+  /** Root of the Expo project directory */
+  projectDir: string;
+  /** Resolved Expo app configuration */
+  exp: ExpoConfig;
+  /**
+   * If we should run in non-interactive mode, as much as we can.
+   * The store authentication requires a fully configured ASC environment, or interactive mode.
+   */
+  nonInteractive: boolean;
+};
+
+export type MetadataAppStoreAuthentication = {
+  /** The root entity of the App store  */
+  app: App;
+  /** The authentication state we used to fetch the root entity */
+  auth: Partial<Session.AuthState>;
+};
+
+/**
+ * Metadata uses the submission profile to find the configured metadata filename.
+ * Note, only iOS is supported for metadata right now.
+ */
+export async function createMetadataContextAsync(params: {
+  projectDir: string;
+  exp: ExpoConfig;
+  profileName?: string;
+  nonInteractive?: boolean;
+}): Promise<MetadataContext> {
+  const submissionProfiles = await getProfilesAsync({
+    type: 'submit',
+    platforms: [Platform.IOS],
+    projectDir: params.projectDir,
+    profileName: params.profileName,
+  });
+
+  const submissionProfile = submissionProfiles.find(profile => profile.platform === Platform.IOS);
+  assert(
+    submissionProfile,
+    'Could not resolve the iOS submission profile, only iOS is supported for metadata'
+  );
+  const iosSubmissionProfile = submissionProfile.profile as IosSubmitProfile;
+
+  const exp = params.exp ?? getExpoConfig(params.projectDir);
+  const user = await ensureLoggedInAsync({ nonInteractive: params.nonInteractive });
+  const bundleIdentifier = await getBundleIdentifierAsync(params.projectDir, exp);
+
+  const credentialsCtx = new CredentialsContext({
```

Comment:
CredentialsContext should not be created in util functions, it should be done in some top-level code e.g. in command, one constraint we want to ensure is that this is not called more than once per command, if this is true then it's fine to leave it here

The other point is it looks like you only need access to apple API, if that is the case you do not need credentials context.

Reply:
> CredentialsContext should not be created in util functions, it should be done in some top-level code e.g. in command, one constraint we want to ensure is that this is not called more than once per command, if this is true then it's fine to leave it here

Alright, thanks for the context! Yeah, we only use the `AppStoreApi` instance to authenticate with the ASC. (see below why I used `CredentialContext` instead of the `AppStoreApi` directly)

> The other point is it looks like you only need access to apple API, if that is the case you do not need credentials context.

I can also initialize the `AppStoreApi` directly, instead of the credentials context. I did this to not go too deep into the inner workings of the credentials. But from your comment I assume you'd prefer that instead :) I'll update it!

Reply:
Done, would need to update the command PR, but I'll do that after merging this and bringing the stacked PRs up-to-date.